platform/macos/src/MGLMapView.mm

```diff
@@ -1695,6 +1698,23 @@ - (void)setRotateEnabled:(BOOL)rotateEnabled {
     _compass.hidden = !rotateEnabled;
 }
 
+- (void)adjustZoomLevelForKey:(NSString *)key {
+    double newZoomLevel = self.zoomLevel;
+    unichar keyUnichar = [key characterAtIndex:0];
+    switch(keyUnichar) {
+        case '-': {
+            newZoomLevel -= 1;
+            break;
+        }
+        case '=':
+        case '+': {
+            newZoomLevel += 1;
+        }
+    }
+
+    [self setZoomLevel:newZoomLevel animated:YES];
```

Comment:
It just occurred to me that we have existing methods that zoom in and out (for ⌘+ and ⌘−, respectively): `-moveToBeginningOfParagraph:` and `-moveToEndOfParagraph:`. By calling those methods (so named because of the ⌥↑ and ⌥↓ key bindings), we’ll also respect the `zoomEnabled` property and the rounding added in #8027.